### DISCUSSION TITLE
The color mapping of the -datacellcol ori command

### DISCUSSION THREAD

**Discussion by LifengGan** [neperfepx/neper#795] [Q&A] [answered]
Dear neper developers,

I want to know if I use `-datacellcol ori` to map the color to crystal orientations. What kind of data will this command use, and is it a random orientation? And does it link to the IPF color key generated by following command from tutorials if I use the cubic system?
```
neper -T -n 1 -domain stdtriangle -dim 2 -o stdtriangle
neper -M stdtriangle.tess -cl 0.02 -statnode col_stdtriangle
neper -V stdtriangle.msh -datanodecol "col:file(stdtriangle.stnode)" -dataeltcol from_nodes -dataelt2dedgerad 0 -dataelt1drad 0.001 -showelt1d all -imagesize 800:400 -print tmp
convert -pointsize 32 -draw "text 180,390 '[001]' text 545,390 '[011]' text 505,30 '[111]'" tmp.png stdtriangle.png
```

or
Is the same orientations data I get from
`neper -T -n 10 -orisampling uniform -crysym cubic -oridescriptor euler-bunge -format ori -o output` 

Thanks very much

Best regards,
Lifeng

**Comment by rquey** [comment, marked as answer]
`-datacellcol ori` will use the orientations of the cells (stored in the `.tess` file) to compute the colors. It will compute them according to the _color scheme_, as defined by `-datacellcolscheme`. For IPF coloring, use [-datacellcolscheme ipf](https://neper.info/doc/neper_v.html#cmdoption-data-cell-poly-face-edge-ver-seed-crystal-vox-colscheme).
